File: passenger-service/src/main/java/io/voitovich/yura/passengerservice/exceptionhandler/model/ExceptionInfo.java
package io.voitovich.yura.passengerservice.exceptionhandler.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.NonNull;
import org.springframework.http.HttpStatus;


@Builder
@Schema(description = "Exception info")
public record ExceptionInfo (
    @Schema(name = "message", description = "error message", example = "Not found")
    @NonNull
    String message,
    @Schema(name = "status", description = "error status", example = "BAD_REQUEST")
    @NonNull
    HttpStatus status
) {}

File: ride-service/src/main/kotlin/io/voitovich/yura/rideservice/exceptionhandler/model/ExceptionInfo.kt
package io.voitovich.yura.rideservice.exceptionhandler.model

import io.swagger.v3.oas.annotations.media.Schema
import org.springframework.http.HttpStatus


@Schema(description = "Represents information about an exception.")
data class ExceptionInfo(
    @Schema(description = "The error status code.", example = "400")
    val code: Int,

    @Schema(description = "The error status.", example = "BAD_REQUEST")
    val status: HttpStatus,

    @Schema(description = "The error message.")
    val message: String
)
